knimin/templates/pm_targeted_pool.html

```diff
@@ -76,79 +59,63 @@
-    updateTotalValues();
-  }
-
-  /**
-   *
-   * Update the computed values when the plate percentage is changed
-   *
-   * @param {Element} input The input element that has been updated
-   *
-   **/
-  function platePercentageUpdated(input) {
-    var currentEntry = $(input).parents('.entry:first');
-    var runTotalPoolVolume = parseFloat($('#run-pool-volume-input').val());
-    var perc = parseFloat($(input).val());
-    // Magic number 10 -> pool-input is in micrograms, so we need
-    // to multiply 1000 to get nanograms (pool plate volume), since
-    // we have to divide by 100 due to the percentage value: 1000/100 = 10s
-    $(currentEntry).find('.pm-plate-pool').val(runTotalPoolVolume * 10 * perc);
-    updateTotalValues();
-  }
-
-  /**
-   *
-   * Update the computed values when the Total pool volume is changes
-   *
-   **/
-  function runTotalPoolVolumeUpdated() {
-    var runTotalPoolVolume = parseFloat($('#run-pool-volume-input').val());
-    $.each($('.entry'), function (idx, entry) {
-      var perc = parseFloat($(entry).find('.pm-percentage').val());
-      // Magic number 10 -> pool-input is in micrograms, so we need
-      // to multiply 1000 to get nanograms (pool plate volume), since
-      // we have to divide by 100 due to the percentage value: 1000/100 = 10s
-      $(entry).find('.pm-plate-pool').val(runTotalPoolVolume * 10 * perc);
-    });
-    updateTotalValues();
-  }
-  /**
-   *
    * Updates the total values in the GUI
    *
    **/
-  function updateTotalValues() {
+  function updateValues() {
     var totalSamples = 0;
     $.each($('.entry .pm-num-samples'), function (idx, input) {
-      totalSamples += parseInt($(input).val());
+      var val = $(input).val();
+      if (val !== "") {
+        totalSamples += parseInt(val);
+      }
     });
     $('#total-samples').val(totalSamples);
 
     var totalPerc = 0;
     $.each($('.entry .pm-percentage'), function (idx, input) {
-      totalPerc += parseInt($(input).val());
+      var numSamples = $(input).parents('.entry:first').find('.pm-num-samples').val();
+      var val = 0;
+      if (numSamples !== "") {
+        val = (parseInt(numSamples) / totalSamples) * 100;
+      }
+      $(input).val(val);
+      totalPerc += val;
     });
     $('#total-percentage').val(totalPerc);
 
-    var totalVol = 0;
-    $.each($('.entry .pm-plate-pool'), function (idx, input) {
-      totalVol += parseInt($(input).val());
+    var totalDNAConc = 0;
+    $.each($('.entry .pm-dna-conc'), function (idx, input) {
+      var val = $(input).val();
+      if (val !== "") {
+        totalDNAConc += parseFloat($(input).val());
+      }
+    });
+    $('#total-dna-conc').val(totalDNAConc);
+
+    var total5Amt = 0;
+    $.each($('.entry .pm-5-amt'), function (idx, input) {
+      var perc = parseFloat($(input).parents('.entry:first').find('.pm-percentage').val());
+      // This is how the value of this column was computed in the Excel file
```

Comment:
Verified! This specific computation was correct but I was missing the normalization value.